cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/lineage/LineageOperationsProcessor.java

```diff
@@ -47,8 +48,16 @@
  * converting it into the form the platform expects. This includes prefixing operation names
  * to make sure they are unique across all the plugins, adding implicit merge operations when
  * stage has more than one input etc.
+ * The platform operation requires each operation to specify the previous operation name and the operation name has
+ * to be unique. Therefore this class will prefix each operation name with the stage name, i.e,
+ * {stage-name}.{original-name}. When a stage has multiple input stages except Joiner, implicit merge operations are
+ * generated in order for future stages to look up the previous operations.
+ *
+ * TODO: CDAP-16395 revisit this class to see if we can get rid of the merge operations since user will be confused
+ * about it. If we cannot, should just make this class more understandable.
```

Comment:
we should get rid of them. Multiple inputs going into a non-joiner is a no-op since nothing changes in the data. It should not generate a new operation.

Reply:
Yea, we should remove it, but in 6.2 it is too risky. It is better we do it early in next release.